rq/job.py

```diff
@@ -1580,13 +1580,15 @@ def __init__(self, max: int, interval: Union[int, List[int]] = 0):
 
 
 class Callback:
-    def __init__(self, func: Callable[..., Any], timeout: Optional[Any] = None):
-        if not inspect.isfunction(func) and not inspect.isbuiltin(func):
-            raise ValueError('Callback func must be a function')
+    def __init__(self, func: Union[str, Callable[..., Any]], timeout: Optional[Any] = None):
+        if not isinstance(func, str) and not inspect.isfunction(func) and not inspect.isbuiltin(func):
+            raise ValueError('Callback func must be a string or function')
 
         self.func = func
         self.timeout = parse_timeout(timeout) if timeout else CALLBACK_TIMEOUT
 
     @property
     def name(self) -> str:
+        if isinstance(self.func, str):
+            return self.func
```

Comment:
@rishabh-ranjan you need to add a test where `Callback` is passed in with a string.

Reply:
please take a look at the new commit. may need some polish, but is hopefully most of the way there.